cpu/cc2538/dev/sys-ctrl.h

```diff
@@ -242,6 +244,37 @@
 #endif
 /** @} */
 /*---------------------------------------------------------------------------*/
+/** \name System clock divisor selection
+ * @{
+ */
+#ifdef SYS_CTRL_CONF_SYS_DIV
+#if SYS_CTRL_CONF_SYS_DIV & ~SYS_CTRL_CLOCK_CTRL_SYS_DIV
+#error Invalid system clock divisor
+#endif
+#define SYS_CTRL_SYS_DIV		SYS_CTRL_CONF_SYS_DIV
+#else
+#define SYS_CTRL_SYS_DIV		SYS_CTRL_CLOCK_CTRL_SYS_DIV_16MHZ
+#endif
+
+#ifdef SYS_CTRL_CONF_IO_DIV
+#if SYS_CTRL_CONF_IO_DIV & ~SYS_CTRL_CLOCK_CTRL_IO_DIV
+#error Invalid I/O clock divisor
+#endif
+#define SYS_CTRL_IO_DIV			SYS_CTRL_CONF_IO_DIV
+#else
+#define SYS_CTRL_IO_DIV			SYS_CTRL_CLOCK_CTRL_IO_DIV_16MHZ
+#endif
+
+/* Returns actual system clock in Hz */
+#define SYS_CTRL_SYS_CLOCK		(SYS_CTRL_32MHZ >> SYS_CTRL_SYS_DIV)
+/* Returns actual I/O clock in Hz */
+#define SYS_CTRL_IO_CLOCK		(SYS_CTRL_32MHZ >> (SYS_CTRL_IO_DIV >> 8))
+
+#if SYS_CTRL_IO_CLOCK > (5 * SYS_CTRL_SYS_CLOCK / 3)
+#error I/O clock must not exceed 5/3 of system clock
+#endif	
```

Comment:
This check and the empty line above it can be removed now that we keep the UART on the system clock.